src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -864,14 +876,22 @@ void M()
             switch (analysisScope)
             {
                 case BackgroundAnalysisScope.ActiveFile:
-                    workspace.OpenDocument(document.Id);
+                    if (isSourceGenerated)
+                        workspace.OpenSourceGeneratedDocument(document.Id);
+                    else
+                        workspace.OpenDocument(document.Id);
```

Comment:
See other comments about not splitting up the APIs in TestWorkspace -- it seems to me to be easier just to keep them unified unless there's a reason I'm missing here. Otherwise this is just adding code here to call one of two methods that assert the test code called the right method.

Reply:
➡️ This is calling a method on `TestWorkspace`. Discussion is covered under https://github.com/dotnet/roslyn/pull/58363#discussion_r785242974.